### PR TITLE
Removed custom MQTT client

### LINKED ISSUE #45: Generic-lamp node creating mqtt flood
I am not sure what logs I can submit for this one. It also might be usual behavior, however, it's causing me issues.

I can force the issue when I hit deploy in node-red. By using MQTT.fx I subscribe to the node's messages and it floods out about 100 messages, they are all the same "{"linkquality":45,"state":"ON"}". 
The thing is I have about 10 nodes for all my lights and they all send ~100 messages. This then queues any new messages I send. So if I hit deploy then press my hue dimmer button it can take up to 15 seconds to send the command. It always works, you just have to wait. 

Throughout the day this seems to occur, a storm of messages that will delay a command I am trying to send. I end up pressing it a few times then all at once they come in once the message flood has finished. 

Versions:
node-red-contrib-zigbee2mqtt-devices: 0.15.2
node-red: 1.2.3
mosquitto: 1.6.10

### LINKED ISSUE #66: Remove bridge config from generic-lamp
The bridge config is completely redundant in some nodes, as it must also be configured in the device-config.

It is used and can be removed in the following nodes:
- generic-lamp
- get-lamp-state

### PR DESCRIPTION
**Pull request by Dirnei**
#### Features:

- Removed custom MQTT Configuration and replaced it with the MQTT Configuration from the core nodes
- Added option to `bridge-config` to disable device refresh on deployment
- Added option to `bridge-config` to log log-messages from zigbee2mqtt to the `debug tab`
- Removed unnecessary `bridge-config` from `generic-lamp` and `get-lamp-state` configuration. Resolves #66 

#### Bug fixes:

- Deployment increased the refresh messages that will be sent on deployment. Fixes #45
- Redone the device-list as payload has changed and this topic is now also retained on the mqtt broker it should solve #32

### CODE REVIEW

File: src/lib/mqtt.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+export class MqttSubscription {
+    topic: string;
+    regex: RegExp;
+    jsonPayload: boolean;
+    callback: MqttSubscriptionCallback;
+
+    constructor(topic: string, jsonPayload: boolean, callback: MqttSubscriptionCallback) {
+        this.topic = topic;
+        let tmp: string = topic.split("+").join("[^\/]*");
```
Comment (Andreas-Huber) [inline src/lib/mqtt.ts:12]: Please add a comment and maybe a sample of what this regex/string manipulation does.
I don't want to think about it.

File: src/lib/mqtt.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    invokeIfMatch(topic: string, payload: any) {
+        if (this.comapreTopic(topic)) {
+            if (this.jsonPayload) {
+                try{
+                    payload = JSON.parse(payload);
+                } catch(err){
+                    console.log("################################################################");
+                    console.log(" node-red-contrib-zigbee2mqtt-devices error:");
+                    console.log(err);
```
Comment (Andreas-Huber) [inline src/lib/mqtt.ts:27]: Should these logs be a `log.error()`?

File: src/lib/mqtt.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+                    console.log("################################################################");
+                }
+            }
+
+            this.callback(payload, topic);
+        }
+    }
+
+    comapreTopic(topic: string) {
```
Comment (Andreas-Huber) [inline src/lib/mqtt.ts:36]: Specify return type in methods?
`comapreTopic(topic: string) : boolean {`
`invokeIfMatch(topic: string, payload: any) : void {`

File: src/nodes/devices-eurotronic.js [hunk 4]
```diff
@@ -29,7 +29,7 @@ module.exports = function (RED) {
         }
 
         utils.setConnectionState(bridgeNode, node);
-        bavaria.observer.register(bridgeNode.id + "_connected", function (message) {
+        let regId = bavaria.observer.register(bridgeNode.id + "_connected", function (message) {
```
Comment (Andreas-Huber) [inline src/nodes/devices-eurotronic.js:32]: I believe these can be made const in all files.... If you want.

File: src/zigbee2mqtt.js [hunk 5]
```diff
@@ -63,7 +63,7 @@ module.exports = function (RED) {
             });
         }
 
-        bavaria.observer.register(bridgeNode.id + "_connected", function (_msg) {
+        let regId1 = bavaria.observer.register(bridgeNode.id + "_connected", function (_msg) {
```
Comment (Andreas-Huber) [inline src/zigbee2mqtt.js:66]: Can be made const. regId2 as well.

**Review by Andreas-Huber — CHANGES_REQUESTED** [review]
Done with code. Still have to test it.

File: src/nodes/devices-eurotronic.js (hunk 4 above)
Comment (Dirnei) [reply src/nodes/devices-eurotronic.js:32]: That's what I've done. If you see one I don't changed, let me know.

Comment (Dirnei) [reply src/nodes/devices-eurotronic.js:32]: Sorry I skipped the word const while reading ^^

**Review by Andreas-Huber — APPROVED** [review]
Tested the upgrade. The MQTT broker has to be reconfigured once - that should be fine.

**Pull request merged**